### ISSUE TITLE
VPN cannot be turned ON on MacOS

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#2425]
**VPN version:**

* 2.7.0 (2.202112061823);

**Affected Platforms**:

* MacOS; 

**Prerequisites**:

* Have a FxA account with an active subscription for Mozilla VPN;
* Have Mozilla VPN installed and open;
* Be signed in;

**Steps to reproduce**:
1. Turn on the VPN;
2. Observe the behaviour;

**Expected result**:

* The VPN is successfully turned on;

**Actual result**:

* The VPN cannot connect. The toggle switches to Disconnecting and OFF;

**Notes**:

* Reproducible on Production and Staging servers;
* Not reproducible on other platforms; 
* This is a blocker issue, because the VPN cannot be turned ON on MacOS;
* See the attached video;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-1435)

**Comment by data-sync-user** [comment]
➤ Luminita Buruian commented:

Owen Kirby Sure thing.

I have reproduced this on Mozilla VPN 2.7.0 (2.202112091222), Qt5 build. Please find the logs attached.

Let me know if you need anything else.

Thanks.

CC: Madalin Cotetiu Valentina Virlics 

[^mozillavpn-2021-12-9.txt]

**Comment by data-sync-user** [comment]
➤ Owen Kirby commented:

I have been able to reproduce the issue by installing the MacOS client built from Github CI, but when built locally it appears to just work. The root cause of the problem here seems to be that the wireguard-go binary has not been correctly signed, and this leads to the tunnel failing to start and exiting with a crash. The smoking gun in the logs appears to be: Warning: (macos - WireguardUtilsMacos) Tunnel process encountered an error: QProcess::Crashed

Checking out the MacOS console also shows that the wireguard-go binary is failing to start due to a code-signing problem.

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

Owen, can you tell me more? What do you see when you run:
`codesign --verify --verbose /Applications/Mozilla\ VPN.app/Contents/Resources/utils/wireguard-go`

**Comment by oskirby** [comment]
Here are my findings:
```
okirby@okirby-pllywp mozilla-vpn-client % codesign --verify --verbose /Applications/Mozilla\ VPN.app/Contents/Resources/utils/wireguard-go 
/Applications/Mozilla VPN.app/Contents/Resources/utils/wireguard-go: valid on disk
/Applications/Mozilla VPN.app/Contents/Resources/utils/wireguard-go: satisfies its Designated Requirement
```

```
okirby@okirby-pllywp mozilla-vpn-client % codesign -dv --verbose=4 /Applications/Mozilla\ VPN.app/Contents/Resources/utils/wireguard-go
Executable=/Applications/Mozilla VPN.app/Contents/Resources/utils/wireguard-go
Identifier=wireguard-go
Format=Mach-O thin (x86_64)
CodeDirectory v=20500 size=32440 flags=0x10000(runtime) hashes=1003+7 location=embedded
VersionPlatform=1
VersionMin=658944
VersionSDK=786432
Hash type=sha256 size=32
CandidateCDHash sha256=e0d77d5cffd48004da37bade33a5fe3e400ca5d5
CandidateCDHashFull sha256=e0d77d5cffd48004da37bade33a5fe3e400ca5d5325e37ee4a2b30efb58dcf48
Hash choices=sha256
CMSDigest=e0d77d5cffd48004da37bade33a5fe3e400ca5d5325e37ee4a2b30efb58dcf48
CMSDigestType=2
Executable Segment base=0
Executable Segment limit=2793472
Executable Segment flags=0x1
Page size=4096
CDHash=e0d77d5cffd48004da37bade33a5fe3e400ca5d5
Signature size=8983
Authority=Developer ID Application: Andrea Marchesini (7W87BT2W9M)
Authority=Developer ID Certification Authority
Authority=Apple Root CA
Timestamp=Dec 13, 2021 at 4:08:52 AM
Info.plist=not bound
TeamIdentifier=7W87BT2W9M
Runtime Version=12.0.0
Sealed Resources=none
Internal requirements count=1 size=172
```

And the MacOS console reports the following when trying to start the connection:
```
Process:               wireguard-go [14127]
Path:                  /Applications/Mozilla VPN.app/Contents/Resources/utils/wireguard-go
Identifier:            wireguard-go
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        Mozilla VPN [14108]
Responsible:           Mozilla VPN [14108]
User ID:               0

Date/Time:             2021-12-13 15:59:47.519 -0800
OS Version:            macOS 11.6 (20G165)
Report Version:        12
Bridge OS Version:     5.5 (18P4759a)
Anonymous UUID:        94705BE6-6B8F-20BF-5AD4-55B59D26E606

Sleep/Wake UUID:       B34A4AD8-D857-4216-905F-2CE6A1306ED4

Time Awake Since Boot: 32000 seconds
Time Since Wake:       1900 seconds

System Integrity Protection: enabled

Crashed Thread:        Unknown

Exception Type:        EXC_CRASH (Code Signature Invalid)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace CODESIGNING, Code 0x1

kernel messages:

VM Regions Near 0 (cr2):
--> 
    mapped file                   1000000-12aa000      [ 2728K] r-x/rwx SM=COW  Object_id=4bd22825

Backtrace not available

Unknown thread crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000000  rcx: 0x0000000000000000  rdx: 0x0000000000000000
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x0000000000000000  rsp: 0x00007ffeefbffd00
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x0000000000000000
  rip: 0x000000000548f000  rfl: 0x0000000000000200  cr2: 0x0000000000000000
  
Logical CPU:     0
Error Code:      0x00000000
Trap Number:     0

Unknown thread instruction stream not available.

Unknown thread last branch register state not available.


Binary images description not available


External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
Writable regions: Total=8608K written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=8608K(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
STACK GUARD                       56.0M        1 
Stack                             8192K        1 
VM_ALLOCATE                          8K        2 
VM_ALLOCATE (reserved)             432K        3         reserved VM address space (unallocated)
mapped file                       4012K        7 
===========                     =======  ======= 
TOTAL                             68.3M       14 
TOTAL, minus reserved VM space    67.9M       14

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.7.0 (2.202112150959), while using MacOS Big Sur.

The VPN is successfully turned on.
